devicetypes/qubino/qubino-flush-2-relay.src/qubino-flush-2-relay.groovy

```diff
@@ -0,0 +1,451 @@
+	}
+	child?.sendEvent(map)
+	createEvent(map)
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd, ep) {
+	log.warn "Unhandled ${cmd}" + (ep ? " from endpoint $ep" : "")
+}
+
+def on() {
+	onOffCmd(0xFF)
+}
+
+def off() {
+	onOffCmd(0x00)
+}
+
+def ping() {
+	refresh()
+}
+
+def childOnOff(deviceNetworkId, value) {
+	def switchId = getSwitchId(deviceNetworkId)
+	if (switchId != null) sendHubCommand onOffCmd(value, switchId)
+}
+
+private onOffCmd(value, endpoint = 1) {
+	delayBetween([
+			encap(zwave.basicV1.basicSet(value: value), endpoint),
+			encap(zwave.basicV1.basicGet(), endpoint),
+	])
+}
+
+def childRefresh(deviceNetworkId, includeMeterGet = true) {
+	def switchId = getSwitchId(deviceNetworkId)
+	if (switchId != null) {
+		sendHubCommand refresh([switchId],includeMeterGet)
+	}
+}
+
+def refresh(endpoints = [1], includeMeterGet = true) {
+
+	def cmds = []
+
+	endpoints.each {
+		cmds << [encap(zwave.basicV1.basicGet(), it)]
+		if (includeMeterGet) {
+			cmds << encap(zwave.meterV3.meterGet(scale: 0), it)
+			cmds << encap(zwave.meterV3.meterGet(scale: 2), it)
+		}
+	}
+
+	delayBetween(cmds, 200)
+}
+
+private resetAll() {
+	childDevices.each {
+		if (it.deviceNetworkId != state.temperatureSensorDni) {
+			childReset(it.deviceNetworkId)
+		}
+	}
+	sendHubCommand reset()
+}
+
+def childReset(deviceNetworkId) {
+	def switchId = getSwitchId(deviceNetworkId)
+	if (switchId != null) {
+		log.debug "Child reset switchId: ${switchId}"
+		sendHubCommand reset(switchId)
+	}
+}
+
+def reset(endpoint = 1) {
+	log.debug "Resetting endpoint: ${endpoint}"
+	delayBetween([
+			encap(zwave.meterV3.meterReset(), endpoint),
+			encap(zwave.meterV3.meterGet(scale: 0), endpoint),
+			"delay 500"
+	], 500)
+}
+
+def getSwitchId(deviceNetworkId) {
+	def split = deviceNetworkId?.split(":")
+	return (split.length > 1) ? split[1] as Integer : null
+}
+
+private encap(cmd, endpoint = null) {
+	if (cmd) {
+		if (endpoint) {
+			cmd = zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint: endpoint).encapsulate(cmd)
+		}
+
+		if (zwaveInfo.zw.contains("s")) {
+			zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+		} else {
+			cmd.format()
+		}
+	}
+}
+
+private addChildSwitches(numberOfSwitches) {
+	for (def endpoint : 2..numberOfSwitches) {
+		try {
+			String childDni = "${device.deviceNetworkId}:$endpoint"
+			def componentLabel = device.displayName[0..-2] + "${endpoint}"
+			addChildDevice("smartthings", "Child Metering Switch", childDni, device.getHub().getId(), [
+					completedSetup	: true,
+					label			: componentLabel,
+					isComponent		: false
+			])
+		} catch(Exception e) {
+			log.warn "Exception: ${e}"
+		}
+	}
+}
+
+private addChildTemperatureSensor() {
+	try {
+		String childDni = "${device.deviceNetworkId}:${state.numberOfSwitches + 1}"
+		state.temperatureSensorDni = childDni
+		addChildDevice("qubino", "Child Temperature Sensor", childDni, device.getHub().getId(), [
```

Comment:
I had to rename this DTH because of conflicts. Even if they're in different namespaces they. can't have the same name. It's called Qubino Temperature Sensor now.